presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -79,13 +79,58 @@ public void testDistinctLimitOverInequalityJoin()
                 anyTree(
                         node(DistinctLimitNode.class,
                                 anyTree(
-                                        join(INNER, ImmutableList.of(), Optional.of("O_ORDERKEY < L_ORDERKEY"),
-                                                tableScan("orders", ImmutableMap.of("O_ORDERKEY", "orderkey")),
-                                                any(tableScan("lineitem", ImmutableMap.of("L_ORDERKEY", "orderkey"))))
+                                        filter("O_ORDERKEY < L_ORDERKEY",
+                                                join(INNER, ImmutableList.of(), Optional.empty(),
+                                                        tableScan("orders", ImmutableMap.of("O_ORDERKEY", "orderkey")),
+                                                        any(tableScan("lineitem", ImmutableMap.of("L_ORDERKEY", "orderkey"))))
+                                                        .withExactOutputs(ImmutableList.of("O_ORDERKEY", "L_ORDERKEY")))))));
+
+        assertPlan("SELECT DISTINCT o.orderkey FROM orders o JOIN lineitem l ON o.shippriority = l.linenumber AND o.orderkey < l.orderkey LIMIT 1",
+                anyTree(
+                        node(DistinctLimitNode.class,
+                                anyTree(
+                                        join(INNER,
```

Comment:
I do not think it is important for this test. 
Actually maybe we should drop filter from the other query?